server/app/forms/FileUploadQuestionForm.java

```diff
@@ -1,20 +1,56 @@
 package forms;
 
+import java.util.OptionalInt;
 import services.question.types.FileUploadQuestionDefinition;
+import services.question.types.QuestionDefinitionBuilder;
 import services.question.types.QuestionType;
 
 /** Form for updating a file upload question. */
 public class FileUploadQuestionForm extends QuestionForm {
+  private OptionalInt maxFiles;
+
   public FileUploadQuestionForm() {
     super();
+    maxFiles = OptionalInt.empty();
   }
 
   public FileUploadQuestionForm(FileUploadQuestionDefinition qd) {
     super(qd);
+    maxFiles = qd.getMaxFiles();
+  }
+
+  public OptionalInt getMaxFiles() {
+    return maxFiles;
+  }
+
+  /**
+   * We use a string parameter here so that if the field is empty (i.e. unset), we can correctly set
+   * to an empty OptionalInt. Since the HTML input is type "number", we can be sure this string is
+   * in fact an integer when we parse it. If we instead used an int here, we see an "Invalid value"
+   * error when binding the empty value in the form.
+   */
+  public void setMaxFiles(String maxFiles) {
+    if (maxFiles.isEmpty()) {
+      this.maxFiles = OptionalInt.empty();
+    } else {
+      Integer maxFilesInt = Integer.parseInt(maxFiles);
+      this.maxFiles = maxFilesInt <= 0 ? OptionalInt.empty() : OptionalInt.of(maxFilesInt);
```

Comment:
Going with setting the minValue because I think it's definitely ambiguous what a user wants if they enter 0 here so I'd rather not allow it, however I'm not 100% happy with it since that's just a client-side protection, and I'd rather not persist any bad values. I don't think we have precedent for throwing errors from this code so I think it's probably fine / not worth the extra effort to protect against (in general, I'm not 100% sure how this is populated, looks like some play framework magic perhaps?) Any code that reads this is going to handle 0 and negative just fine, anyway.